contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -342,17 +463,253 @@ abstract contract BaseHarvester is Governable {
             address[] memory rewardTokenAddresses = strategy
                 .getRewardTokenAddresses();
 
-            for (uint256 j = 0; j < rewardTokenAddresses.length; j++) {
+            for (uint256 j = 0; j < rewardTokenAddresses.length; ++j) {
                 _swap(rewardTokenAddresses[j], _rewardTo);
             }
         }
     }
 
     /**
-     * @dev Swap a reward token for stablecoins on Uniswap. The token must have
-     *       a registered price feed with the price provider.
-     * @param _swapToken Address of the token to swap.
+     * @dev Swap a reward token for the base token on the configured
+     *      swap platform. The token must have a registered price feed
+     *      with the price provider
+     * @param _swapToken Address of the token to swap
      * @param _rewardTo Address where to send the share of harvest rewards to
      */
-    function _swap(address _swapToken, address _rewardTo) internal virtual;
+    function _swap(address _swapToken, address _rewardTo) internal virtual {
```

Comment:
Rather than return the swap function, I just changed it to do the actual swap to keep it a bit simple. But let me know if you feel there's a better way

Reply:
This works.